src/App.jsx

```diff
@@ -14,45 +15,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreperedGoods(goods, { sortField, reverse }) {
+  const preperedGoods = goods.map((good, i) => (
+    {
+      name: good,
+      length: good.length,
+      id: i + 1,
+    }
+  ));
+
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1[sortField].localeCompare(good2[sortField]);
+
+        case SORT_FIELD_LENGTH:
+          return good1[sortField] - good2[sortField];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    return [...preperedGoods].reverse();
```

Comment:
u don`t need this spread oeprator, after map u already have  copy of array

Reply:
> u don`t need this spread oeprator, after map u already have copy of array

Thank you, I will definitely use it next time!